packages/alchemy/src/middleware/simulate-uo.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  deepHexlify,
+  resolveProperties,
+  type UserOperationStruct,
+} from "@alchemy/aa-core";
+import type { AlchemyProvider } from "../provider/base.js";
+import type { ClientWithAlchemyMethods } from "./client.js";
+
+export const withAlchemyUserOpSimulation = <P extends AlchemyProvider>(
+  provider: P
+): P => {
+  provider.withSimulateUOMiddleware(async (uoStruct) => {
+    const uoSimResult = await (
```

Comment:
if `P extends AlchemyProvider` and `AlchemyProvider` has a method on it now called `simulateUserOperationAssetChanges` let's just do `const uoSimResult = simulateUserOperationAssetChange`

Reply:
only reason I didn't is because `simulateUserOperationAssetChanges` builds the uoStruct (so its input are the same as sendUO for good devex), where is this middleware is part of building the uoStruct and so 1. the inputs are and 2. would lead to an additional unnecessary buildUO call